discovery-provider/src/api/v1/tracks.py

```diff
@@ -275,6 +277,69 @@ def get(self):
         return success_response(trending)
 
 
+# Get random tracks for a genre and exclude tracks in the exclusion list
+random_track_parser = reqparse.RequestParser()
+random_track_parser.add_argument('genre', required=False)
+random_track_parser.add_argument('limit', type=int, required=False)
+random_track_parser.add_argument('exclusion_list', type=int, action='append', required=False)
+random_track_parser.add_argument('time', required=False)
+
+@ns.route("/random")
+class RandomTrack(Resource):
+    @record_metrics
+    @ns.doc(
+        id="""Random Tracks""",
+        params={
+            'genre': 'Random trending tracks for a specified genre',
+            'limit': 'Number of random tracks to fetch',
+            'exclusion_list': 'List of track ids to exclude',
+            'time': 'Trending tracks over a specified time range (week, month, allTime)'
+        },
+        responses={
+            200: 'Success',
+            400: 'Bad request',
+            500: 'Server error'
+        }
+    )
+    @ns.marshal_with(tracks_response)
+    @cache(ttl_sec=TRENDING_TTL_SEC)
+    def get(self):
+        args = random_track_parser.parse_args()
+        limit = format_limit(args, default_limit=DEFAULT_RANDOM_LIMIT)
+        args['limit'] = max(TRENDING_LIMIT, limit)
```

Comment:
couldn't we pass the TRENDING_LIMIT as the `max_limit` arg to `format_limit` to avoid having to do this max line?

Reply:
Great question. The reason why we can't do that is because there are 2 kinds of limits. Line 308 is the limit which the client specified (with the format_limit bounds), aka the final number of tracks to return. Line 309 is the number of recommended tracks we want to fetch, before shuffling and returning a subset with length of first limit (line 308) that user specified.
Hope that makes sense.